pkg/util/ecs/metadata/detection.go

```diff
@@ -137,3 +140,56 @@ func getAgentV4URLFromEnv() (string, error) {
 	}
 	return agentURL, nil
 }
+
+// IsMetadataV4Available checks if the v4 metadata endpoint is available.
+// It checks if metadata v4 url is available in the environment variable.
+// It should be available when the agent is deployed by ECS, as the ECS agent injects this environment variable.
+// Then it checks if agent is deployed directly on EC2 running in ECS.
+// It returns true if any container has the v4 endpoint env variable.
+func IsMetadataV4Available() (bool, error) {
+	// Agent is deployed by ECS
+	_, err := getAgentV4URLFromEnv()
+	if err == nil {
+		return true, nil
+	}
+
+	// If the agent is deployed on Fargate and v4 is not available, return false
+	if config.IsECSFargate() {
+		return false, errors.New("v4 endpoint not found in Fargate")
```

Comment:
it is not clear to me where we are checking here if v4 is available or not?

it looks like we are directly returning an error if we are running on ECS Fargate, while the comment above says that we return false if we are on fargate **and** v4 is not available.

Reply:
This part is deleted when detect metadata v4 by checking ecs agent version